### PR TITLE
Fixes for nosystemd or non-elementary distros

### PR DESCRIPTION
**Pull request by paledega**
logout not working on nosystemd distro. I fixed it. we have secondary logout plan (pkill -KILL -u $USER)

### CODE REVIEW

File: src/Widgets/EndSessionDialog.vala [hunk 1]
```diff
@@ -181,6 +182,8 @@ public class Session.Widgets.EndSessionDialog : Gtk.Window {
                     server.confirmed_logout ();
                     logout_interface.terminate ();
                 } catch (GLib.Error e) {
+                    //I dont know but work :D
+                    Posix.system("pkill -KILL -u $USER");
```
Comment (paledega) [inline src/Widgets/EndSessionDialog.vala:186]: This line run shell command for kill all userspace process (logout)

File: src/Widgets/EndSessionDialog.vala [hunk 2]
```diff
@@ -22,6 +22,7 @@
  * docs taken from unity indicator-session's
  * src/backend-dbus/org.gnome.SessionManager.EndSessionDialog.xml
  */
+using Posix;
```
Comment (paledega) [inline src/Widgets/EndSessionDialog.vala:25]: We need Posix module for running shell command

File: meson.build [hunk 3]
```diff
@@ -11,6 +11,8 @@ prefix = get_option('prefix')
 libdir = join_paths(prefix, get_option('libdir'))
 
 add_global_arguments('-DGETTEXT_PACKAGE="@0@"'.format(gettext_name), language:'c')
+add_global_arguments('--pkg',language:'vala')
+add_global_arguments('posix',language:'vala')
```
Comment (paledega) [inline meson.build:15]: We must add Posix module in meson build file

**Comment by paledega** [comment]
Hmm lint error.

**Comment by frknkrc44** [comment]
looks like work well, but can we check that in gnome-session to find a solution except a shell command? because if user don't have to pkill applet in PATH variable and not using systemd, this solution won't work.

**Comment by cassidyjames** [comment]
elementary OS uses systemd, and maintaining alternate code paths that aren't used on elementary OS that execute POSIX commands seems prone to breakage. I'm inclined to close this; however, a downstream shipping Pantheon on a non-systemd base is welcome to patch this downstream as the see fit.

**Comment by paledega** [comment]
hmm no problem 👍
I created patch. Maintainers can patch source code for fixing nosystemd

**Comment by paledega** [comment]
https://github.com/elementary/wingpanel-indicator-session/issues/122#issuecomment-639015512

**Pull request closed without merging**